### ISSUE TITLE
getDartDevelopmentServiceVersion Method not found

### ISSUE THREAD

**Issue by yauheniprakapenka** [flutter/devtools#2648]
Good day.

When I click on the Dev Tools button, a disconnect occurs. I tried:
- to run ios and android in VS Code
- to run android in android studio
- Reinstalled dart
- Flutter clean
- Removed app from device

How to solve this problem?

Thanks

Installed extensions in VS Code:
Flutter 3.19.0
Dart 3.19.1

```
[✓] Flutter (Channel stable, 1.22.6, on Mac OS X 10.15.7 19H2 darwin-x64, locale en-RU)
    • Flutter version 1.22.6 at /Users/yauheni.prakapenka/flutter
    • Framework revision 9b2d32b605 (7 days ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/yauheni.prakapenka/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/yauheni.prakapenka/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.8.4

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin installed
    • Dart plugin version 201.9317
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin installed
    • Dart plugin version 203.6912

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.19.0

[✓] Connected device (1 available)
    • AOSP on IA Emulator (mobile) • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)
    ! Error: iPhone Xr is not connected. Xcode will continue when iPhone Xr is connected. (code -13)
```

![Screenshot 01-30-2021 11 08 15](https://user-images.githubusercontent.com/47568606/106351270-556ddf80-62ec-11eb-9ae7-543b9ac309ac.png)

**Comment by Alieas-PC** [comment]
same issue, any updates?

**Comment by ArizArmeidi** [comment]
same issue here, I have tried reinstalling flutter and dart but still can't get the devtools to work

**Comment by ItoYo16u** [comment]
I found this commit cause this bug.
https://github.com/flutter/devtools/commit/bdd2a799ebecc427110451255bbc0143eb504b9e#diff-1a6177a61dfab2e28e6f3589c6a3ccf898014651ce23a63d4ad465ead59b3595R42


https://github.com/dart-lang/sdk/blob/master/pkg/dds/lib/vm_service_extensions.dart

**Comment by yuke27** [comment]
same issue. How to solve this problem?

**Comment by barak-innov** [comment]
I have the same issue!

**Comment by indyant** [comment]
I've got the exactly same issue on both Windows 10 and macos 11.1. 
Development environment is mostly same as @yauheniprakapenka
Is there any workaround to it?

**Comment by ItoYo16u** [comment]
try switch to dev channel

**Comment by indyant** [comment]
> Workaround: switch to dev channel.
> 
> ```shell
> flutter channel dev
> ```

It works. 
I'll stick to the dev channel until next stable version is released. 

Thanks.

**Comment by bkonyi** [comment]
Duplicate of https://github.com/flutter/devtools/issues/2649. Should have a fix released in 0.9.7+1 shortly.

**Comment by daniel-v** [comment]
> Duplicate of #2649. Should have a fix released in 0.9.7+1 shortly.

Not quite! I tried to activate it but version resolution failed:

```
flutter pub global activate devtools 0.9.7+1
Resolving dependencies...
Because devtools >=0.9.7+1 depends on devtools_shared 0.9.7+1 which doesn't match any versions, devtools >=0.9.7+1 is forbidden.
So, because pub global activate depends on devtools 0.9.7+1, version solving failed.
```

**Comment by bkonyi** [comment]
> > Duplicate of #2649. Should have a fix released in 0.9.7+1 shortly.
> 
> Not quite! I tried to activate it but version resolution failed:
> 
> ```
> flutter pub global activate devtools 0.9.7+1
> Resolving dependencies...
> Because devtools >=0.9.7+1 depends on devtools_shared 0.9.7+1 which doesn't match any versions, devtools >=0.9.7+1 is forbidden.
> So, because pub global activate depends on devtools 0.9.7+1, version solving failed.
> ```

Yeah, I ran into a snag trying to publish `devtools_shared` and `devtools_app` and I'm waiting on another team member to come back and give me permissions to publish. I'll post here when I've been able to finish publishing the packages successfully. Thanks for your patience!

**Comment by bkonyi** [comment]
We're having some trouble getting hold of contributors with the right permissions to publish. I'll do my best to have the fix published as soon as I can, but in the meantime you should be able to work around the failure by falling back to an older version of DevTools: `flutter pub global activate devtools 0.9.6+3`.

**Comment by daniel-v** [comment]
Awesome response time! I appreciate you keeping us updated so regularly.

What you proposed, using an older version was my first attempt. Unfortunately, either Android Studio or something checks if I'm using the latest and greatest. If not, on first start, it updates to 0.9.7. At least that's what I observed.

**Comment by bkonyi** [comment]
> Awesome response time! I appreciate you keeping us updated so regularly.
> 
> What you proposed, using an older version was my first attempt. Unfortunately, either Android Studio or something checks if I'm using the latest and greatest. If not, on first start, it updates to 0.9.7. At least that's what I observed.

No problem! I try my best :-)

Are you comfortable with the command line? You might want to try starting `devtools` manually from there (I think `flutter pub global run devtools` should work?). If that doesn't work and Android Studio decides it wants to upgrade you again, you might have to check out the last working version of DevTools and run your app using `flutter run`. Hopefully that's enough to unblock you until we get this sorted out!

**Comment by daniel-v** [comment]
Yeah, I get by in command line (thanks for not assuming I'm a bashguru). 

I'm currently not working on any production applications, just playing around, therefore not really blocked (swapping to dev channel works just fine till I can get back to stable). I'll defer trying your last tip to someone else.

I did check though: currently (on dev channel), devtools is ran like so:

```
flutter pub global run devtools --no-launch-browser
```

**Comment by getgizmo** [comment]
For people using VScode, is there versions of Flutter or Dart that we should use to fallback on versus running "flutter pub global activate devtools 0.9.6+3" as this does not work for me

**Comment by youssefali424** [comment]
nothing is currently working i deactivated it and activated it again with a lower version still same error and I used command line

**Comment by bkonyi** [comment]
Duplicate of https://github.com/flutter/devtools/issues/2648. This should be resolved in the latest version (`0.9.7+1`). Thanks for being patient while we tried to get this resolved!

**Comment by jeiea** [comment]
Is this really fixed? I got same error with `0.9.7+1`.

**Comment by bkonyi** [comment]
@jeiea are you having issues just trying to connect? If so, can you share some logs from the browser console for me to look at tomorrow? This should be resolved in 0.9.7+1 as there's only one place we call `getDartDevelopmentServiceVersion()` and we should be catching the exception there as of the last release.

**Comment by jeiea** [comment]
```
Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
main.dart.js:54844 PreferencesController: storage not initialized
main.dart.js:54844 DevTools version 0.9.7.
main.dart.js:54854 getDartDevelopmentServiceVersion: (-32601) Method not found

log0 @ main.dart.js:54854
DevTools failed to load SourceMap: Could not load content for http://localhost:8000/sourcemaps/content.js.map: 
```

Not so much information, but it's all. I copied `build` directory from `0.9.7` due to #2653, please consider it.

**Comment by bkonyi** [comment]
> ```
> Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
> main.dart.js:54844 PreferencesController: storage not initialized
> main.dart.js:54844 DevTools version 0.9.7.
> main.dart.js:54854 getDartDevelopmentServiceVersion: (-32601) Method not found
> 
> log0 @ main.dart.js:54854
> DevTools failed to load SourceMap: Could not load content for http://localhost:8000/sourcemaps/content.js.map: 
> ```
> 
> Not so much information, but it's all. I copied `build` directory from `0.9.7` due to #2653, please consider it.

That was the issue from `0.9.7` that lead to us publishing `0.9.7+1` which seemed to go wrong somehow... :-) I've only just gotten the permissions I need to re-publish myself, which should hopefully resolve the issue.

**Comment by bkonyi** [comment]
If you try a `pub global activate devtools` or `flutter pub global activate devtools` now, you'll pull down `0.9.7+2` which should work!

**Comment by aboutlan** [comment]
@bkonyi  0.9.7+2 worked for me, everyone could try this version. Thanks bkonyi

**Comment by daniel-v** [comment]
Hey, @bkonyi 

wanted to express my thanks for how you went about keeping track/informing us about what's up. I don't know where this issue went wrong but talking to me as I was beeping through the weekend gave me an appreciation for the team working on this pkg.

**Comment by bkonyi** [comment]
Thanks @daniel-v! :-) We do our best!

**Comment by stkuwanda** [comment]
> If you try a `pub global activate devtools` or `flutter pub global activate devtools` now, you'll pull down `0.9.7+2` which should work!

This worked for me. thanks.

**Comment by sunnykinger** [comment]
> If you try a `pub global activate devtools` or `flutter pub global activate devtools` now, you'll pull down `0.9.7+2` which should work!

0.9.7 was supposed to have a fix for network tab request and response headers empty data, but the problem is still there. Request and Response Headers are empty.

**Comment by aytunch** [comment]
I have the same disconnect issue happening in Flutter 2.5.0
When I try to get in to a slow frame's detail, suddenly I get the disconnect error.
I can not profile using the VScode dev tools. Please recommend a solution.

**Comment by shreetheja** [comment]
> > Workaround: switch to dev channel.
> > ```shell
> > flutter channel dev
> > ```
> 
> It works. I'll stick to the dev channel until the next stable version is released.
> 
> Thanks.

When will this be fixed??

**Comment by RehanQazi76** [comment]
> Workaround: switch to dev channel.
> 
> ```shell
> flutter channel dev
> ```

**Comment by RehanQazi76** [comment]
> Workaround: switch to dev channel.
> 
> ```shell
> flutter channel dev
> ```

tried this buddy , but still have the same issue

**Comment by bkonyi** [comment]
This issue has been resolved, so I'm going to lock this thread. If you're still experiencing issues on the latest version of Flutter, please file a new issue with `flutter doctor` output and reproduction steps. Thanks!
